daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Inputs.scala

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf.transaction
+
+import com.daml.lf.data.Ref
+import com.daml.lf.value.Value.ContractId
+
+import scala.collection.immutable.TreeSet
+
+/** The inputs of the transaction, that is:
+  *   - the contracts were fetched
```

Comment:
What about contracts that were returned by a `lookupByKey`? And what about contracts that are exercised, but not fetched?